src/engraving/dom/transpose.cpp

```diff
@@ -833,9 +833,15 @@ void Score::transpositionChanged(Part* part, Interval oldV, Fraction tickStart,
                             n->undoChangeProperty(Pid::TPC2, tpc);
                         }
                     }
-                    for (Note* n : c->notes()) {
-                        int tpc = transposeTpc(n->tpc1(), v, true);
-                        n->undoChangeProperty(Pid::TPC2, tpc);
+                    const std::vector<Note*> notes = c->notes();
```

Comment:
Isn't `transpositionChanged` about fixing up the score when the user selects a different instrument with a different transposition? I don't think that should depend on the selection filter, should it?